### PR TITLE
Allow to keep typographic whitespace.

### PR DESCRIPTION
**Pull request by sha-red**
Implemented solution for #13.

### CODE REVIEW

File: html_sanitizer/sanitizer.py [hunk 1]
```diff
@@ -34,14 +56,20 @@ def normalize_overall_whitespace(html):
...
+            "&nbsp;",
+            "&#160;",
+            "&#xa0;",
+        ]
+    if whitespace_re is None:
+        whitespace_re = r"\s+"
     for ch in whitespace:
         html = html.replace(ch, " ")
-    html = re.sub(r"(?u)\s+", " ", html)
+    html = re.sub(whitespace_re, " ", html)
```
Comment (matthiask) [inline html_sanitizer/sanitizer.py]: Aren't we losing the normalization of unicode spaces here?

File: html_sanitizer/sanitizer.py [hunk 2]
```diff
@@ -25,7 +23,31 @@ def sanitize_href(href):
...
+    "PUNCTUATION SPACE",
+    "THIN SPACE",
+    "HAIR SPACE",
+    "NARROW NO-BREAK SPACE",
+    "MEDIUM MATHEMATICAL SPACE",
+    "IDEOGRAPHIC SPACE"
+]
+
+typographic_whitespace = "".join({unicodedata.lookup(n)
+    for n in typographic_whitespace_names})
```
Comment (matthiask) [inline html_sanitizer/sanitizer.py]: Minor nit: This does not look as if it was formatted by black. Run `tox` inside the repo to run `black` and `flake8` on the code.

Comment (sha-red) [reply html_sanitizer/sanitizer.py]: Ok, did it.

File: html_sanitizer/sanitizer.py (hunk 1 above)
Comment (sha-red) [reply html_sanitizer/sanitizer.py]: Yeah, you're right at least for Python 2 (shouldn't make a difference in Python 3 as strings are handled as Unicode anyway). I'm inserting the Unicode flag into the regex for Python 2 now.

**Comment by matthiask** [comment]
Thanks, looks good!

**Comment by matthiask** [comment]
Released as 1.7.3

**Pull request merged**